src/services/api.js

```diff
@@ -1,17 +1,44 @@
-export async function getProducts(params = {}) { 
+export async function getProducts(params = {}) {
   const query = new URLSearchParams(params).toString();
 
   try {
-    const response = await fetch(
-      `https://panda-market-api.vercel.app/products?${query}`
-    );
+    const response = await fetch(`https://panda-market-api.vercel.app/products?${query}`);
     if (!response.ok) {
       throw new Error(`HTTP error: ${response.status}`);
     }
     const body = await response.json();
     return body;
   } catch (error) {
-    console.error("Failed to fetch products:", error);
+    console.error('Failed to fetch products:', error);
+    throw error;
+  }
+}
+
+export async function getProductById(productId) {
+  try {
+    const response = await fetch(`https://panda-market-api.vercel.app/products/${productId}`);
```

Comment:
p4;
api 요청시 사용되는 기본 서버 url은 상수로 관리해도 좋을 것 같아요!

```suggestion
    const response = await fetch(`${API_BASE_URL}/products/${productId}`);
```